bascule/basculehttp/metrics.go

```diff
@@ -0,0 +1,48 @@
+package basculehttp
+
+import (
+	"net/http"
+
+	"github.com/Comcast/comcast-bascule/bascule"
+)
+
+type Listener interface {
+	OnAuthenticated(bascule.Authentication)
+}
+
+type listenerDecorator struct {
+	listener Listener
+}
+
+func (m *listenerDecorator) decorate(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(response http.ResponseWriter, request *http.Request) {
+		ctx := request.Context()
+		auth, ok := bascule.FromContext(ctx)
+		if !ok {
+			response.WriteHeader(http.StatusForbidden)
+			return
+		}
+		if m.listener != nil {
+			m.listener.OnAuthenticated(auth)
+		}
+		next.ServeHTTP(response, request)
+
+	})
+}
+
+type LOption func(*listenerDecorator)
+
+func WithMeasures(listener Listener) LOption {
```

Comment:
`WithMeasures` isn't a good name.  `WithListener` or something similar would be better.

Additionally, it makes a better API to allow multiple listeners here.  In `golang` you must mention the package.  So if you only allow (1) listener per option, you end up with something noisy:

```
basculehttp.NewListenerDecorator(
    basculehttp.WithListener(a),
    basculehttp.WithListener(b),
    basculehttp.WithListener(c),
)
```

Contrast that with allowing multiple listeners in a single option, which reduces noise:

```
basculehttp.NewListenerDecorator(
    basculehttp.WithListeners(a, b, c),
)
```